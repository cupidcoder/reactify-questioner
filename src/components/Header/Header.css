.Header {
  display: flex;
  height: 7rem;
  padding: 1rem;
  background-color: #f39e3e;
  justify-content: space-between;
}

.Logo {
  color: #ffffff;
}

.Logo h1 {
  font-family: 'Tangerine', cursive;
  font-size: 4rem;
}

.Nav ul {
  list-style: none;
}

.Nav ul li {
  color: #ffffff;
  font-size: 2rem;
}

.Nav ul li:not(:last-child) {
  border-right: 1px solid #202C39;
}
