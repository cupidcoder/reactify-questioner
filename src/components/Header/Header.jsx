import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import NavBar from '../NavBar/NavBar';
import style from './Header.css';

const Header = ({ token }) => {
  let items = [
    { no: 1, text: 'Sign in', path: '/signin' },
    { no: 2, text: 'Sign up', path: '/signup' }
  ];
  const isLoggedIn = () => {
    if (token) {
      items = [
        { no: 1, text: 'Profile', path: '#' },
        { no: 2, text: 'Logout', path: '#' }
      ];
    }
  };
  isLoggedIn();
  return (
    <header>
      <div className={style.Header}>
        <div className={style.Logo}>
          <h1>
            <Link to='/'>
              Questioner
            </Link>
          </h1>
        </div>
        <div className={style.Nav}>
          <NavBar items={items} />
        </div>
      </div>
    </header>
  );
};

const mapStateToProps = (state) => {
  if (state.auth.data) {
    return { token: state.auth.data.token };
  }
  return { token: undefined };
};

const HeaderComponent = connect(mapStateToProps)(Header);

export { Header, HeaderComponent };
