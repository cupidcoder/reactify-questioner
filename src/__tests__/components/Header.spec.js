import React from 'react';
import { shallow } from 'enzyme';
import { Header } from '../../components/Header/Header';

describe('Header', () => {
  it('Should correctly save snapshot of the Header', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
  });
});
