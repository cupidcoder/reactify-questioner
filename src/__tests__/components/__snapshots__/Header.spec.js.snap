// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Should correctly save snapshot of the Header 1`] = `ShallowWrapper {}`;
